python/metatensor-operations/metatensor/operations/sort.py

```diff
@@ -0,0 +1,237 @@
+    ... )
+    >>> print(block)
+    TensorBlock
+        samples (3): ['structures', 'centers']
+        components (): []
+        properties (3): ['n', 'l']
+        gradients: None
+    >>> block_sorted_stepwise = metatensor.sort_block(block, axes=["properties"])
+    >>> # properties are are sorted which are columns in this case
+    >>> block_sorted_stepwise.values
+    array([[2, 0, 1],
+           [5, 3, 4],
+           [8, 6, 7]])
+    >>> block_sorted_stepwise = metatensor.sort_block(
+    ...     block_sorted_stepwise, axes=["samples"]
+    ... )
+    >>> # samples are are sorted which are rows in this case
+    >>> block_sorted_stepwise.values
+    array([[5, 3, 4],
+           [8, 6, 7],
+           [2, 0, 1]])
+    >>> sorted_block = metatensor.sort_block(block)
+    >>> np.all(
+    ...     sorted_block.values == block_sorted_stepwise.values
+    ... )  # doing everything at once and verify correctness
+    True
+    >>> # Example for sorting the gradient
+    >>> sorted_block.add_gradient(
+    ...     parameter="g",
+    ...     gradient=TensorBlock(
+    ...         values=np.arange(18).reshape(3, 2, 3),
+    ...         samples=Labels(["sample"], np.array([[1], [2], [0]])),
+    ...         components=[Labels.range("direction", 2)],
+    ...         properties=sorted_block.properties,
+    ...     ),
+    ... )
+    >>> sorted_grad_block = metatensor.sort_block(sorted_block)
+    >>> sorted_grad_block.gradient("g").samples == Labels.range("sample", 3)
+    True
+    >>> sorted_grad_block.gradient("g").properties == sorted_block.properties
+    True
+    >>> sorted_grad_block.gradient(
+    ...     "g"
+    ... ).values  # sorting of components can be clearly seen
+    array([[[12, 13, 14],
+            [15, 16, 17]],
+    <BLANKLINE>
+           [[ 0,  1,  2],
+            [ 3,  4,  5]],
+    <BLANKLINE>
+           [[ 6,  7,  8],
+            [ 9, 10, 11]]])
+
+    """
+    if isinstance(axes, str):
+        if axes == "all":
+            axes_list = ["samples", "components", "properties"]
+        elif axes not in ["samples", "components", "properties"]:
+            raise ValueError(
+                f"input parameter 'axes' may only be one of the strings 'samples', "
+                f"components' or 'properties' but is '{axes}'"
+            )
+        else:
+            axes_list = [axes]
+    elif isinstance(axes, list):
+        for axis in axes:
+            if axis not in ["samples", "components", "properties"]:
+                raise ValueError(
+                    f"input parameter 'axes' may only contain the strings 'samples', "
+                    f"components' or 'properties' but contains '{axis}'"
+                )
+        axes_list = axes
+    else:
+        raise TypeError("input paramater 'axes' should be of type str or list of str")
+
+    result_block = _sort_single_block(block, axes_list, descending)
+
+    # issue #68
+    # for parameter, gradient in block.gradients().items():
```

Comment:
```suggestion
```

Not sure what this means? Where you referring to https://github.com/lab-cosmo/metatensor/pull/309?

Reply:
yes, exactly I forgot about this, adapted